crates/core/tedge_api/src/workflow/toml_config.rs

```diff
@@ -0,0 +1,482 @@
+                }
+                "restart" => {
+                    // TODO impl a clean From<TomlExitHandlers> implementation
+                    let on_exec: GenericStateUpdate = input
+                        .handlers
+                        .on_exec
+                        .map(|u| u.into())
+                        .unwrap_or_else(|| "executing".to_string().into());
+                    let on_success: GenericStateUpdate = input
+                        .handlers
+                        .on_success
+                        .map(|u| u.into())
+                        .unwrap_or_else(|| "successful".to_string().into());
+                    let on_error: GenericStateUpdate = input
+                        .handlers
+                        .on_error
+                        .map(|u| u.into())
+                        .unwrap_or_else(|| "failed".to_string().into());
+
+                    Ok(OperationAction::Restart {
+                        on_exec: on_exec.status,
+                        on_success: on_success.status,
+                        on_error: on_error.status,
+                    })
+                }
+                _ => Err(WorkflowDefinitionError::UnknownAction { action: command }),
+            },
+        }
+    }
+}
+
+impl TryFrom<TomlOperationWorkflow> for OperationWorkflow {
+    type Error = WorkflowDefinitionError;
+
+    fn try_from(input: TomlOperationWorkflow) -> Result<Self, Self::Error> {
+        let operation = input.operation;
+        let handlers = input.handlers.try_into()?;
+        let mut states = HashMap::new();
+        for (state, action_spec) in input.states.into_iter() {
+            let action = TryInto::<OperationAction>::try_into(action_spec)?;
+            states.insert(state, action);
+        }
+
+        Ok(OperationWorkflow {
+            operation,
+            built_in: false,
+            handlers,
+            states,
+        })
+    }
+}
+
+/// User-Friendly representation of an [ExitHandlers]; as used in the operation TOML definition files
+///
+/// A user don't have to give a handler for all possible exit code.
+/// - A handler can simply be a string used as the next state for the command.
+/// - A handler can be attached to a range of exit code
+/// - A wildcard handler can be defined as a default handler
+/// - `on_success` is syntactic sugar for `on_exit.0`
+/// - `on_error` is syntactic sugar for `on_exit._`
+///
+/// Some combinations are not valid and are rejected when the operation model is built from its TOML representation.
+/// - `on_success` and `on_exit.0` are are synonyms and cannot be both provided
+/// - `on_error` and `on_exit._` are are synonyms and cannot be both provided
+/// - `on_success` and `on_stdout` are incompatible, as the next state is either determined from the script stdout or its exit codes
+/// - `on_exec` is only meaningful in the context of a background script or a builtin action
```

Comment:
Sure. But this would be more something along what is proposed here: https://github.com/thin-edge/thin-edge.io/pull/2496/files#diff-0bc2ab86d1dc663436d9764c1ee3b48333013f3d52ce410aac0af84705f0bf7cR364

.i.e. to have two states.

```toml
["agent-restart"]
background_script = "sudo systemctl restart tedge-agent"
on_exec = "waiting-for-restart"

["waiting-for-restart"]
script = "/some/script.sh checking restart"
on_success = "successful_restart"
on_error =  "failed_restart"
on_timeout = "waiting-for-restart"
```